components/homme/src/theta-l_kokkos/cxx/EquationOfState.hpp

```diff
@@ -81,22 +81,19 @@ class EquationOfState {
     // To avoid temporaries, use exner to store some temporaries
     ColumnOps::compute_midpoint_delta(kv,phi_i,exner);
 
-    Kokkos::parallel_for(Kokkos::ThreadVectorRange(kv.team,NUM_LEV),
-                         [&](const int ilev) {
-#ifndef NDEBUG
+    int nerr;
+    Kokkos::parallel_reduce(Kokkos::ThreadVectorRange(kv.team,NUM_LEV),
+                            [&](const int ilev, int& nerr) {
       // check inputs
       const int vec_len = (ilev==(NUM_LEV-1)) ? ColInfo<NUM_PHYSICAL_LEV>::LastPackLen : VECTOR_SIZE;
       for (int iv=0; iv<vec_len; ++iv) {
-        if (vtheta_dp(ilev)[iv]<0.0) {
-          Kokkos::abort("Error! vtheta_dp>0 detected.\n");
-        }
-        if (exner(ilev)[iv]>0.0) {
-          Kokkos::abort("Error! dphi>0 detected.\n");
-        }
+        if (vtheta_dp(ilev)[iv] < 0.0 || exner(ilev)[iv] > 0.0)
```

Comment:
If may consider moving this check inside the `compute_pnh_and_exner` fcn below (and make it return nerr or a bool), and remove the check in the DirkFunctor, since they would both use the same. Or not.

Reply:
I had thought about it but decided I don't like the idea of our lowest-level routines that do raw computations being polluted with always-on checks.